spec.html

```diff
@@ -197,6 +295,75 @@ <h1>The %TypedArray% Intrinsic Object</h1>
+                        1. <del>Assert: Both Type(_x_) and Type(_y_) are Number or both are BigInt.</del>
+                        1. <del>If _comparefn_ is not *undefined*, then</del>
+                          1. <del>Let _v_ be ? ToNumber(? Call(_comparefn_, *undefined*, &laquo; _x_, _y_ &raquo;)).</del>
+                          1. <del>If IsDetachedBuffer(_buffer_) is *true*, throw a *TypeError* exception.</del>
+                          1. <del>If _v_ is *NaN*, return *+0*<sub>𝔽</sub>.</del>
+                          1. <del>Return _v_.</del>
+                        1. <del>If _x_ and _y_ are both *NaN*, return *+0*<sub>𝔽</sub>.</del>
+                        1. <del>If _x_ is *NaN*, return *1*<sub>𝔽</sub>.</del>
+                        1. <del>If _y_ is *NaN*, return *-1*<sub>𝔽</sub>.</del>
+                        1. <del>If _x_ &lt; _y_, return *-1*<sub>𝔽</sub>.</del>
+                        1. <del>If _x_ &gt; _y_, return *1*<sub>𝔽</sub>.</del>
+                        1. <del>If _x_ is *-0*<sub>𝔽</sub> and _y_ is *+0*<sub>𝔽</sub>, return *-1*<sub>𝔽</sub>.</del>
+                        1. <del>If _x_ is *+0*<sub>𝔽</sub> and _y_ is *-0*<sub>𝔽</sub>, return *1*<sub>𝔽</sub>.</del>
+                        1. <del>Return *+0*<sub>𝔽</sub>.</del>
+                        1. <ins>Return ? CompareTypedArrayElements(_x_, _y_, _comparefn_, _buffer_).</ins>
+                      1. <del>Return ? SortIndexedProperties(_obj_, _len_, _SortCompare_).</del>
+                      1. <ins>Let _sortedList_ be ? SortIndexedProperties(_obj_, _len_, _SortCompare_, *true*).</ins>
+                      1. <ins>Let _j_ be 0.</ins>
+                      1. <ins>Repeat, while _j_ &lt; _len_,</ins>
+                        1. <ins>Perform ! Set(_obj_, ! ToString(𝔽(_j_)), _sortedList_[_j_], *true*).</ins>
+                        1. <ins>Set _j_ to _j_ + 1.</ins>
+                      1. <ins>Return _obj_.</ins>
+                    </emu-alg>
+                </emu-clause>
+
+                <emu-clause id="sec-comparetypedarrayelements" type="abstract operation">
+                    <h1>
+                    CompareTypedArrayElements(
+                        _x_: unknown,
+                        _y_: unknown,
+                        _comparefn_: a function object or *undefined*,
+                        _buffer_: An ArrayBuffer
+                    ): either a normal completion containing a Number or an abrupt completion
+                    </h1>
+                    <dl class="header">
+                    </dl>
+                    <emu-alg>
+                        1. Assert: Both Type(_x_) and Type(_y_) are Number or both are BigInt.
+                        1. If _comparefn_ is not *undefined*, then
+                            1. Let _v_ be ? ToNumber(? Call(_comparefn_, *undefined*, &laquo; _x_, _y_ &raquo;)).
+                            1. If IsDetachedBuffer(_buffer_) is *true*, throw a *TypeError* exception.
```

Comment:
Does this need to be checked? Isn't it done implicitly with the `Get` and `Set` calls?

Reply:
Nope, Get and Set don't fail on detached buffers.

But there's the question of whether it _should_ be checked here, and I think the answer is no. The current implementation - and most other TA methods - checks only at the start of the method, and does not re-check after every possible point at which it might get detached.

Reply:
Good point. This check was there right from the [ES6 version](https://262.ecma-international.org/6.0/#sec-%typedarray%.prototype.sort) of `TypedArray.prototype.sort`.

However until ES2021 sorting was defined as performing:

> an implementation-dependent sequence of calls to the [Get](https://262.ecma-international.org/11.0/#sec-get-o-p), [Set](https://262.ecma-international.org/11.0/#sec-set-o-p-v-throw) ...

So the buffer could become detached before reading had finished. This was updated to match the current semantics in https://github.com/tc39/ecma262/pull/1585

Reply:
Yeah, probably.

Let's move discussion of what the behavior _should_ be to https://github.com/tc39/ecma262/pull/2646#issuecomment-1067456576 (which is about a similar change to a different TA method). Changing the existing behavior is out of scope for this proposal, I imagine.